identity/src/main/java/com/android/identity/Util.java

```diff
@@ -298,15 +298,40 @@ static boolean cborDecodeBoolean(@NonNull byte[] data) {
         return simple.getSimpleValueType() == SimpleValueType.TRUE;
     }
 
+    /**
+     * Accepts a {@code DataItem}, attempts to cast it to a {@code Number}, then returns the value
+     * Throws {@code IllegalArgumentException} if the {@code DataItem} is not a {@code Number}. This
+     * method also checks bounds, and if the given data item is too large to fit in a long, it
+     * throws {@code ArithmeticException}.
+     */
+    static long checkedLongValue(DataItem item) {
+        final BigInteger bigNum = castTo(Number.class, item).getValue();
+        final long result = bigNum.longValue();
+        if (!bigNum.equals(BigInteger.valueOf(result))) {
+            throw new ArithmeticException(
+                "Overflow: Unable to convert integer of bit length "
+                    + bigNum.bitLength()
```

Comment:
I think you forgot to update the exception message, it still quotes bitLength?

How about

"Expected long value, got: " + bigNum